src/python/platforms/android/adb.py

```diff
@@ -147,9 +167,32 @@ def run():
   return bytes_output.strip().decode('utf-8', errors='ignore')
 
 
+def copy_file(filename, src_dir, dest_dir, timeout=None):
+  """Copy file between two locations"""
+  file_src = os.path.join(src_dir, filename)
+  if environment.is_android_cuttlefish():
+    cvd_address = get_gce_address()
+    if not cvd_address:
+      logs.log_fatal_and_exit(
+          'Unable to get cvd ip address on cuttlefish host')
+
+    if file_src.endswith('.tar.gz'):
+      cmd = f'ssh {cvd_address} tar xzvf - < {file_src}'
```

Comment:
We shouldn't be unpacking in a function called "copy_file". If we want to unpack, this should be a separate function

Since this is only called in the cuttlefish context, we just need the one case here: 

```python
def copy_to_cuttlefish(src_path, dest_path, timeout=None)
  """Copy file between to the cuttlefish host"""
  cvd_address = get_gce_address()
  return execute_command(f'scp -R {src_path} {cvd_address}:{dest_path}', timeout=timeout)
```

I think you also need the "-r" to make directory copying work.

As below, can we untar in `download_latest_build` instead and avoid doing it here?